rls/src/main/java/io/grpc/rls/AdaptiveThrottler.java

```diff
@@ -193,9 +193,9 @@ private static final class Slot {
       volatile long count;
       // The nearest 0 modulo slot boundary in nanoseconds. The slot boundary
       // is exclusive. [previous_slot.end, end)
-      final long endNanos;
+      final Long endNanos;
```

Comment:
Looking at this change, I don't see the point of NULL_SLOT any more. It seems we should just use `null` for empty slots and all the `currentSlot.endNanos != null` checks would just become `currentSlot != null`.

I do think the code could keep the idea of null slot without the checks by allocating a null slot based on `ticker`.

```java
    TimeBasedAccumulator(long internalNanos, Ticker ticker) {
      ...
      this.slotNanos = internalNanos / NUM_SLOTS;
      this.currentIndex = 0;
      Slot nullSlot = new Slot(ticker.read() - slotNanos - 1);
      for (int i = 0; i < NUM_SLOTS; i++) {
        slots.set(i, nullSlot);
      }
```

Either way seems fine to me.